### PR TITLE
fix divideBy exp/metabolite term issues

### PR DESCRIPTION
**Pull request by congyu-lu**
## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/plots/matrix.config.js [hunk 1]
```diff
@@ -240,7 +240,7 @@ export function setComputedConfig(config) {
 			})
 	}
 	for (const f of config.legendValueFilter.lst) {
-		if (!f.legendGrpName || !f.tvs?.term?.type.startsWith('gene')) continue
+		if (!f.legendGrpName || !f.tvs?.term?.type.startsWith('geneVariant')) continue
```
Comment (xzhou82) [inline client/plots/matrix.config.js:243]: == rather than startsWith?

File: client/plots/matrix.controls.js [hunk 2]
```diff
@@ -145,6 +146,10 @@ export class MatrixControls {
 				},
 				processInput: tw => {
 					if (tw) fillTermWrapper(tw, this.opts.app.vocabApi)
+					if (tw?.term?.type == TermTypes.GENE_EXPRESSION || tw?.term?.type == TermTypes.METABOLITE_INTENSITY) {
```
Comment (xzhou82) [inline client/plots/matrix.controls.js:149]: in matrix grouping by any numeric term should always be discrete, then can do isNumericTerm(tw?.term) tw.q.mode=discrete

and in what case is tw missing in this function?

please add all above comments in function

File: client/plots/matrix.interactivity.js [hunk 3]
```diff
@@ -1607,10 +1607,15 @@ function setSampleGroupActions(self) {
 				// the filter for the categorical or survival term exist, but the current legend key is not there.
 				self.config.legendValueFilter.lst[filterGrpIndex].tvs.values.push({ key: self.activeLabel.grp.id })
 			}
-		} else if (term.type == 'integer' || term.type == 'float') {
+		} else if (
```
Comment (xzhou82) [inline client/plots/matrix.interactivity.js:1610]: replace with isNumericTerm()

File: client/plots/matrix.renderers.js [hunk 4]
```diff
@@ -358,8 +359,17 @@ export function setRenderers(self) {
 		g.append('title').text(`${cl.Samples} are grouped by this gene or variable. Click to edit.`)
 
 		const customMenuOptions = []
-		const tvsKey = ['integer', 'float'].includes(self.config.divideBy.term.type) ? 'ranges' : 'values'
-		if (self.config.legendValueFilter.lst?.find(l => l.legendGrpName == self.config.divideBy.id)?.tvs[tvsKey]?.length) {
+		const tvsKey = ['integer', 'float', TermTypes.GENE_EXPRESSION, TermTypes.METABOLITE_INTENSITY].includes(
```
Comment (xzhou82) [inline client/plots/matrix.renderers.js:362]: replace with isNumericTerm

File: server/src/termdb.matrix.js [hunk 5]
```diff
@@ -153,12 +153,14 @@ async function getSampleData(q) {
...
-						? compute_bins(tw.q, () => {
-								return { min, max }
-						  })
-						: tw.q.lst
+				if (tw.q.type == 'regular-bin') {
+					lst = compute_bins(tw.q, () => {
+						return { min, max }
+					})
+					for (const b of lst) {
+						if (!('name' in b) && b.label) b.name = b.label
```
Comment (xzhou82) [inline server/src/termdb.matrix.js:161]: what's purpose of doing this?

File: client/plots/matrix.controls.js (hunk 2 above)
Comment (congyu-lu) [reply client/plots/matrix.controls.js:155]: tw missing when dividedBy term deleted.
Added comment

File: server/src/termdb.matrix.js (hunk 5 above)
Comment (congyu-lu) [reply server/src/termdb.matrix.js:161]: Adding to compute_bins will break many binning tests. In **_termdb.sql.numeric.js_**, name was added the same as above in getCTE. Maybe we could update compute_bins in another PR?

Comment (xzhou82) [reply server/src/termdb.matrix.js:161]: sure let's fix it in new pr

**Review by xzhou82 — APPROVED** [review]

**Pull request merged**